### PR TITLE
Add autoupdate resources config/version

### PR DESCRIPTION
**Pull request by vapopov**
In this PR implemented new resources for autoupdate configuration and version

Related: https://github.com/gravitational/teleport/pull/39805
changelog: Added resources for autoupdate (config and version)

### CODE REVIEW

File: api/proto/teleport/autoupdate/v1/autoupdate.proto [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+package teleport.autoupdate.v1;
+
+import "teleport/header/v1/metadata.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/autoupdate/v1;autoupdate";
+
+// AutoupdateConfig is a config singleton used to configure cluster
+// autoupdate settings.
+message AutoupdateConfig {
```
[the remark this answers is not in the document]
Comment (sclevine) [reply api/proto/teleport/autoupdate/v1/autoupdate.proto:25]: After discussion: going with `AutoUpdate` in code, but `autoupdate_`/`autoupdate-` elsewhere

File: api/proto/teleport/legacy/client/proto/event.proto
[the remark this answers is not in the document]
Comment (vapopov) [reply api/proto/teleport/legacy/client/proto/event.proto:1]: I've added to this PR

**Review by rosstimothy — APPROVED** [review]

**Review by hugoShaka — APPROVED** [review]

File: api/proto/teleport/legacy/client/proto/event.proto [hunk 2]
```diff
@@ -177,5 +178,9 @@ message Event {
     teleport.machineid.v1.SPIFFEFederation SPIFFEFederation = 62;
     // StaticHostUser is a resource for static host users.
     teleport.userprovisioning.v1.StaticHostUser StaticHostUser = 63;
+    // AutoUpdateConfig is a resource for autoupdate config.
+    teleport.autoupdate.v1.AutoUpdateConfig AutoupdateConfig = 64;
+    // AutoUpdateVersion is a resource for autoupdate version.
+    teleport.autoupdate.v1.AutoUpdateVersion AutoupdateVersion = 65;
```
Comment (sclevine) [inline api/proto/teleport/legacy/client/proto/event.proto:184]:
```suggestion
    // AutoUpdateConfig is a resource for autoupdate config.
    teleport.autoupdate.v1.AutoUpdateConfig AutoUpdateConfig = 64;
    // AutoUpdateVersion is a resource for autoupdate version.
    teleport.autoupdate.v1.AutoUpdateVersion AutoUpdateVersion = 65;
```

**Review by sclevine — APPROVED** [review]

File: docs/pages/includes/helm-reference/zz_generated.teleport-kube-agent.mdx [hunk 3]
```diff
@@ -1499,7 +1499,7 @@ for more information.
 | `object` | `{}` |
 
 `extraLabels.pod` are labels to set on the Pods created by the
-Deployment or StatefulSet.
+Deployment, StatefulSet, or Job.
```
Comment (hugoShaka) [inline docs/pages/includes/helm-reference/zz_generated.teleport-kube-agent.mdx:1502]: Can you undo this change? Master has been broken today and [the fix is on the way](https://github.com/gravitational/teleport/pull/46243).

**Review by hugoShaka — CHANGES_REQUESTED** [review]

File: docs/pages/includes/helm-reference/zz_generated.teleport-kube-agent.mdx (hunk 3 above)
Comment (vapopov) [reply docs/pages/includes/helm-reference/zz_generated.teleport-kube-agent.mdx:1502]: sure, was typing a message to you already :)

**Review by hugoShaka — APPROVED** [review]
